presto-native-execution/src/test/java/com/facebook/presto/nativeworker/SymlinkManifestGeneratorUtils.java

```diff
@@ -0,0 +1,160 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.hive.HiveType;
+import com.facebook.presto.hive.metastore.Column;
+import com.facebook.presto.tpch.ColumnNaming;
+import com.google.common.collect.ImmutableList;
+import io.airlift.tpch.TpchColumn;
+import io.airlift.tpch.TpchEntity;
+import io.airlift.tpch.TpchTable;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.tpch.TpchMetadata.getPrestoType;
+
+/**
+ * Utility class for generating and managing symlink manifests tables.
+ * This class provides methods to create symlink manifest files, clean up symlink data,
+ * and handle TPCH-related table columns and types.
+ */
+public final class SymlinkManifestGeneratorUtils
+{
+    private static final Logger log = Logger.get(SymlinkManifestGeneratorUtils.class);
+    private SymlinkManifestGeneratorUtils() {}
+
+    /**
+     * Creates a symlink manifest file at the specified directory.
+     * This method collects all file paths under the hiveTableLocation and writes them
+     * to a symlink manifest file named "symlink_manifest" inside the specified symlinkManifestDir.
+     *
+     * @param hiveTableLocation the root directory of the Hive table files
+     * @param symlinkManifestDir the directory where the symlink manifest should be created
+     * @throws IOException if an I/O error occurs while creating the directories or writing the manifest
+     */
+    public static void createSymlinkManifest(Path hiveTableLocation, Path symlinkManifestDir) throws IOException
+    {
+        if (Files.notExists(symlinkManifestDir.getParent())) {
+            Files.createDirectory(symlinkManifestDir.getParent());
+        }
+
+        if (Files.notExists(symlinkManifestDir)) {
+            Files.createDirectory(symlinkManifestDir);
+        }
+
+        Path manifestFilePath = symlinkManifestDir.resolve("symlink_manifest");
+        try (BufferedWriter writer = Files.newBufferedWriter(manifestFilePath, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)) {
+            List<String> fileList = new ArrayList<>();
+            collectDataFiles(hiveTableLocation, fileList);
+
+            for (String filePath : fileList) {
+                writer.write(filePath);
+                writer.newLine();
```

Comment:
This method is platform specific and should be avoided. Write the particular line end character you want such as \n

https://docs.oracle.com/javase/8/docs/api/java/io/BufferedWriter.html#newLine--